src/cronjobs

```diff
@@ -116,3 +116,51 @@ p({
     }
   """
 })
+
+p({
+  "class":"foam.nanos.cron.Cron",
+  "id":"CleanupTempFileCron",
+  "description":"Cron to clean up files which exist more than 4 hours",
+  schedule:{
+    class:"foam.nanos.cron.IntervalSchedule",
+    duration:{
+      class:"foam.nanos.cron.TimeHMS",
+      hour:1,
+      minute:0,
+      seconds:0
+    }
+  },
+  "code": """
+    import foam.core.X;
+    import foam.dao.ArraySink;
+    import foam.dao.DAO;
+    import foam.nanos.fs.TempFile;
+    import java.io.File;
+    import java.time.LocalDateTime;
+    import java.time.ZoneId;
+    import java.util.Date;
+    import java.util.List;
+
+    final int TTL = 4;
+
+    DAO tempFileDAO = (DAO) x.get("tempFileDAO");
+
+    Date threshold = Date.from(LocalDateTime.now().minusHours(TTL).atZone(ZoneId.systemDefault()).toInstant());
+
+    List tempFiles = ((ArraySink) tempFileDAO
+      .where(MLang.LT(TempFile.CREATED, threshold))
+      .select(new ArraySink()))
+      .getArray();
+
+    TempFile tempFile;
+
+    for ( int i = 0 ; i < tempFiles.size() ; i++ ) {
+      tempFile = (TempFile) tempFiles.get(i);
+       if ( forHoursBefore.after(tFile.getCreated()) ) {
+        tempFiles.get(i).getFile_().delete();
+        tempFileDAO.remove(tempFile);
+      }
+    }
```

Comment:
Can move the logic for removing file from the file system in a decorator then can just call remove all on the filtered DAO.
```suggestion
    tempFileDAO
      .where(MLang.LT(TempFile.CREATED, threshold))
      .removeAll()
```